### PR TITLE
Password retry out

### PR DESCRIPTION
**Pull request by kojo1**
Zendesk #4966-18

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -3221,14 +3221,24 @@ static int DoUserAuthRequestPassword(WOLFSSH* ssh, WS_UserAuthData* authData,
                 ssh->clientState = CLIENT_USERAUTH_DONE;
                 ret = WS_SUCCESS;
             }
+            else if (ret == WOLFSSH_USERAUTH_INVALID_PASSWORD) {
+                WLOG(WS_LOG_DEBUG, "DUARPW: password check failed");
+                ret = SendUserAuthFailure(ssh, 0);
+            }
```
Comment (ejohnstown) [inline src/internal.c]: Please check for the new authentication failure here and give it its own error code in errors.h.

File: wolfssh/ssh.h [hunk 2]
```diff
@@ -239,6 +239,7 @@ enum WS_UserAuthResults {
     WOLFSSH_USERAUTH_INVALID_AUTHTYPE,
     WOLFSSH_USERAUTH_INVALID_USER,
     WOLFSSH_USERAUTH_INVALID_PASSWORD,
+    WOLFSSH_USERAUTH_PASSWORD_RETRYOUT,
```
Comment (ejohnstown) [inline wolfssh/ssh.h]: Please rename the new authentication code "WOLFSSH_USERAUTH_REJECTED".

File: src/internal.c (hunk 1 above)
Comment (kojo1) [reply src/internal.c]: added WS_USER_AUTH_E

File: wolfssh/ssh.h (hunk 2 above)
Comment (kojo1) [reply wolfssh/ssh.h]: Changed it with WOLFSSH_USERAUTH_REJECTED

File: wolfssh/error.h [hunk 3]
```diff
@@ -105,7 +105,8 @@ enum WS_ErrorCodes {
     WS_CLOSE_FILE_E         = -1065, /* Unable to close local file */
     WS_PUBKEY_REJECTED_E    = -1066, /* Server public key rejected */
     WS_EXTDATA              = -1067, /* Extended Data available to be read */
-
+    WS_USER_AUTH_E          = -1068, /* User authentication error */
+    
     WS_LAST_E               = -1067  /* Update this to indicate last error */
```
Comment (ejohnstown) [inline wolfssh/error.h]: This needs to be -1068.

Comment (ejohnstown) [reply wolfssh/error.h]: This also requires a new error string to be added to GetErrorString() in internal.c.

Comment (kojo1) [reply wolfssh/error.h]: Sorry. It's fixed.

**Pull request merged**